js/views/modals/orderDetail/summaryTab/CompleteOrderForm.js

```diff
@@ -0,0 +1,114 @@
+import $ from 'jquery';
+import {
+  completeOrder,
+  completingOrder,
+  events as orderEvents,
+} from '../../../../utils/order';
+import loadTemplate from '../../../../utils/loadTemplate';
+import Rating from '../../../../models/order/orderCompletion/Rating';
+import BaseVw from '../../../baseVw';
+import RatingsStrip from '../../../RatingsStrip';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    super(options);
+
+    if (!options.model) {
+      throw new Error('Please provide an OrderCompletion model.');
+    }
+
+    if (!options.slug) {
+      throw new Error('Please provide the listing slug.');
+    }
+
+    this.ratingStrips = {};
+    this.slug = options.slug;
+
+    const ratings = this.model.get('ratings');
+
+    if (ratings.length) {
+      this.rating = ratings.at(0);
+    } else {
+      this.rating = new Rating();
+      ratings.push(this.rating);
+    }
+
+    this.listenTo(orderEvents, 'completingOrder', () => {
+      this.getCachedEl('.js-completeOrder').addClass('processing');
+    });
+
+    this.listenTo(orderEvents, 'completeOrderComplete completeOrderFail', () => {
+      this.getCachedEl('.js-completeOrder').removeClass('processing');
+    });
+  }
+
+  className() {
+    return 'completeOrderForm rowLg';
+  }
+
+  events() {
+    return {
+      'click .js-completeOrder': 'onClickCompleteOrder',
+    };
+  }
+
+  onClickCompleteOrder() {
+    const data = {
+      ...this.getFormData(),
+      // If a rating is not set, the RatingStrip view will return 0. We'll
+      // send undefined in that case since it gives us the error message we
+      // prefer.
+      overall: this.ratingStrips.overall.rating || undefined,
+      quality: this.ratingStrips.quality.rating || undefined,
+      description: this.ratingStrips.description.rating || undefined,
+      deliverySpeed: this.ratingStrips.deliverySpeed.rating || undefined,
+      customerService: this.ratingStrips.customerService.rating || undefined,
+      slug: this.slug,
+    };
+
+    this.rating.set(data);
+    this.rating.set(data, { validate: true });
```

Comment:
The reason is that I want to both validate the model and set with potentially invalid values. If you set with the validate option, the model will not set those attributes if any are invalid. But.. since I'm re-rendering the errors, I want them in the model.

Normally, this can be avoided by using Model.save(), which does set the attributes and will also validate, but in this case I don't want that because Save will kick off a request to the server if there are no errors and the request to the server will happen in the utility function.